pkg/server/sessions/internal/transactions/transactions.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2021 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transactions
+
+import (
+	"github.com/codenotary/immudb/embedded/sql"
+	"github.com/codenotary/immudb/pkg/api/schema"
+	"github.com/codenotary/immudb/pkg/database"
+	"sync"
+)
+
+type transaction struct {
+	sync.Mutex
+	transactionID string
+	sqlTx         *sql.SQLTx
+	txMode        schema.TxMode
+	db            database.DB
+	sessionID     string
+}
+
+type Transaction interface {
+	GetID() string
+	GetMode() schema.TxMode
+	Rollback() error
+	Commit() ([]*sql.SQLTx, error)
+	GetSessionID() string
+	SQLExec(request *schema.SQLExecRequest) error
+	SQLQuery(request *schema.SQLQueryRequest) (*schema.SQLQueryResult, error)
+}
+
+func NewTransaction(sqlTx *sql.SQLTx, transactionID string, mode schema.TxMode, db database.DB, sessionID string) *transaction {
```

Comment:
not blocking the PR but I think the NewTransaction should be created from the session struct (assigning the unique id for the tx), but the sqlTx may be created in the constructor of the transaction when executing sqlExec on the db..., no need to start with a sqlTx, it may be just nil

Reply:
is exactly as you described here https://github.com/codenotary/immudb/pull/1014/files/82995df19d329da19a45aec9d3aafbd78a84c93b#diff-6d3b6766c2d825705f2eb1e6fd1e3e84c44c8ab7be2fb28c7190d44247fd0851R70